.github/workflows/CI.yml

```
@@ -33,6 +33,6 @@ jobs:
       - uses: julia-actions/julia-buildpkg@v1
       - uses: julia-actions/julia-runtest@v1
       - uses: julia-actions/julia-processcoverage@v1
-      - uses: codecov/codecov-action@v3
+      - uses: codecov/codecov-action@v4
         with:
```

@ChrisRackauckas this PR breaks codecov integration - if you compare e.g. CI in this PR or the latest commit on the master branch with the second-to-last commit you can see that codecov doesn't run anymore. The problem is that codecov-action@v4 does not support tokenless coverage uploads anymore (see eg the changelog or its README). The token has to be added as a Github secret and then one can use it as
```suggestion
        with:
          token: ${{ secrets.CODECOV_TOKEN }} # required
```